jetty-hazelcast/src/main/java/org/eclipse/jetty/hazelcast/session/HazelcastSessionDataStore.java

```diff
@@ -119,77 +137,62 @@ public boolean isPassivating()
         return true;
     }
 
+    public boolean isFindExpiredSession()
+    {
+        return findExpiredSession;
+    }
+
+    public void setFindExpiredSession( boolean findExpiredSession )
+    {
+        this.findExpiredSession = findExpiredSession;
+    }
+
     @Override
     public Set<String> doGetExpired( Set<String> candidates )
     {
-        if (candidates == null || candidates.isEmpty())
-        {
-            return Collections.emptySet();
-        }
-        
-        long now = System.currentTimeMillis();
-        return candidates.stream().filter( candidate -> {
+        Set<String> expired = new HashSet<>();
+
+        expired.addAll( candidates.stream().filter( candidate -> {
```

Comment:
@janbartel thanks I think it's better you show what you want to achieve because I'm developing based on future changes which are in your mind so it's very difficult for me :-)

Reply:
@olamy ok, have a look at the branch called 'jetty-9.4.x-2609-findexpiredsessions', and the 2
 new methods on AbstractSessionDataStore called doGetOldExpired(long) and cleanOrphans()